components/net/cookie_storage.rs

```diff
@@ -83,7 +83,17 @@ impl CookieStorage {
             Ok(None)
         }
     }
-
+    pub fn clear_storage_for_url(&mut self, url: &ServoUrl) {
+        for (_, cookies) in self.cookies_map.iter_mut() {
```

Comment:
Rather than iterating over every cookie that is stored, we should follow the example of [this code](https://github.com/servo/servo/blob/c35438b4a67255d5e1a88e8ad03894a5cf07f3b6/components/net/cookie_storage.rs#L167-L169), which gets the cookie entry associated with a particular URL. We can then iterate over those cookies only.

Reply:
What does a url mean here ? A document's id or the url which stored this cookie?

Reply:
The document's URL, not the cookie's URL.